anchor/programs/depo/src/states/validation.rs

```diff
@@ -0,0 +1,83 @@
+use crate::states::modules::ConditionModule;
+use crate::states::Escrow;
+use anchor_lang::prelude::*;
+
+use anyhow::Result;
+
+pub trait EscrowContext {
+    fn lamports(&self) -> u64;
+}
+
+impl EscrowContext for Account<'_, Escrow> {
+    fn lamports(&self) -> u64 {
+        **self.to_account_info().lamports.borrow()
+    }
+}
+
+/// Validates escrow modules conditions
+///  
+/// # Arguments
+/// * `escrow` - The escrow account
+/// * `modules` - a slice of ConditionModule to validate
+///
+/// # Returns
+/// * `Result<()>` - Result indicating conditions validation success or failure
+#[allow(dead_code)] // TODO remove this when the method will be used
+pub fn validate_escrow_conditions(
```

Comment:
This method could be called by the release instruction to check if all conditions modules are ok.